components/compass-runtime-agent/internal/compass/director/client.go

```diff
@@ -68,21 +68,58 @@ func (cc *directorClient) FetchConfiguration() ([]kymamodel.Application, error)
 	return applications, nil
 }
 
-func (cc *directorClient) SetURLsLabels(urlsCfg RuntimeURLsConfig) (graphql.Labels, error) {
-	eventsURLLabel, err := cc.setURLLabel(eventsURLLabelKey, urlsCfg.EventsURL)
+func (cc *directorClient) ReconcileLabels(urlsCfg RuntimeURLsConfig) (graphql.Labels, error) {
```

Comment:
you could optimize a bit more by making both queries part of the same http request. Probably the name already used for the client method (FetchConfiguration) is generic enough, so you could include fetching of the runtime labels in this method - doing one http request with two queries in it. Then keep the other client method to be called SetLabels as before and only call it if setting is needed.